docker/setup/terraform/01_provider.tf

```diff
@@ -0,0 +1,17 @@
+# Configure Azure storage to manage Terraform State
+# ref: https://docs.microsoft.com/en-us/azure/developer/terraform/store-state-in-azure-storage
+terraform {
+  backend "azurerm" {
+    resource_group_name   = "tstate"
+    storage_account_name  = "tstate31414"
+    container_name        = "tstate"
+    key                   = "terraform.tfstate"

```

Comment:
Thoughts on making the key dynamic, e.g. based on branch name via the `GITHUB_REF` environment variable, so that we can run this potentially for multiple deployments (e.g. for work-in-progress code) without risking to override the production resources.

Reply:
Sure can! That makes a lot of sense.